src/main/java/lab6/Main.java

```diff
@@ -1,4 +1,145 @@
 package lab6;
 
-public class Main {
+import javafx.application.Application;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+public class Main extends Application {
+    private Button[][] buttons = new Button[3][3];
+    private boolean playerX = true;
+    private Label statusLabel = new Label("Player X's Turn");
+
+    @Override
+    public void start(Stage primaryStage) {
+        // Create a grid layout for the Tic Tac Toe board
+        GridPane grid = new GridPane();
+        grid.setAlignment(Pos.CENTER);
+
+        // Initialize buttons for the game board
+        for (int row = 0; row < 3; row++) {
+            for (int col = 0; col < 3; col++) {
+                Button button = new Button(" ");
+                button.setMinSize(100, 100); // Increase button size for better visibility
+                button.setStyle("-fx-font-size: 30px; -fx-alignment: center;"); // Center align text
+                final int r = row, c = col;
+                button.setOnAction(e -> handleMove(r, c));
+                buttons[row][col] = button;
+                grid.add(button, col, row);
+            }
+        }
+
+        // Reset button to restart the game
+        Button resetButton = new Button("Reset Game");
+        resetButton.setOnAction(e -> resetGame());
+
+        // Layout containing the status label, game grid, and reset button
+        VBox layout = new VBox(15, statusLabel, grid, resetButton);
+        layout.setAlignment(Pos.CENTER);
+
+        // Set up and display the primary stage
+        primaryStage.setScene(new Scene(layout, 400, 500)); // Increased window size
+        primaryStage.setTitle("Lab6: Tic Tac Toe");
+        primaryStage.show();
+    }
+
+    // Handles a player's move when a button is clicked
+    private void handleMove(int row, int col) {
+        if (!buttons[row][col].getText().equals(" ")) return;
+
+        buttons[row][col].setText(playerX ? "X" : "O");
+        if (checkWin()) {
+            showWinMessage(playerX ? "X" : "O");
+        } else if (isBoardFull()) {
+            showDrawMessage();
+        } else {
+            playerX = !playerX;
+            statusLabel.setText("Player " + (playerX ? "X" : "O") + "'s Turn");
+        }
+    }
+
+    // Checks if there is a winning condition on the board
+    private boolean checkWin() {
+        for (int i = 0; i < 3; i++) {
```

Comment:
It would have been nice to see you integrate your Lab 5 code into this rather than going back to a very basic implementation of the game.